### ISSUE TITLE
New AWS service: aws-sdk-sso, is typo protected

### ISSUE THREAD

**Issue by mullermp** [rubygems/rubygems.org#2161]
AWS is launching a new service called SSO (single sign on). When attempting to publish this gem, we get this problem;

```
There was a problem saving your gem: Name 'aws-sdk-sso' is too close to typo-protected gem: aws-sdk-s3
```

We own both gems in this case, so could we get a whitelist for this?

ref: https://github.com/rubygems/rubygems/issues/2978

**Comment by kcrouch** [comment]
I suspect this has blocked about 14,917 processes so far.....

**Comment by cflanny** [comment]
My entire org is blocked by this due to Chef dependencies.  Having to version lock the aws-sdk gems in cookbooks to work around.

**Comment by dwradcliffe** [comment]
@mullermp Go ahead and try now, should be good to go.

**Comment by kcrouch** [comment]
I still can't access aws-sdk-sso via bundler or https://rubygems.org/gems/aws-sdk-sso.  The dependencies for aws-sdk-resources still show a grey "aws-sdk-sso ~> 1" that turns my pointer into a question mark if I hover over it.

**Comment by harshavardhana** [comment]
Please re-open this is not fixed yet.

**Comment by cflanny** [comment]
Agree, still not fixed.

**Comment by obazoud** [comment]
Agree, still not fixed.

**Comment by hsbt** [comment]
Wait for @mullermp comment. Maybe, It's not pushed yet.

**Comment by cjyclaire** [comment]
I'm working on the publishing right now, will update, thanks!

**Comment by cjyclaire** [comment]
Just published, should be resolved now

**Comment by mullermp** [comment]
Beat me to it!

**Comment by dwradcliffe** [comment]
Right now there isn’t a mechanism for that, but we can start to think about how that might work. If you’re interested in contributing maybe you could open an issue with a tech design? 🙂

**Comment by simi** [comment]
@mullermp Do you use one central rubygems.org account at AWS to push the gems or are personal rubygems accounts allowed to push aws gems? I see the latest sso gem was pushed (and is owned) by https://rubygems.org/profiles/awscloud.

I think I can prepare pull request introducing (manually) verified accounts (I think I have shared this idea at some another issue already).

@dwradcliffe WDYT?

**Comment by mullermp** [comment]
Customers are still having issues: https://github.com/aws/aws-sdk-ruby/issues/2159.
We've got the spec dependency - https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-sdk-resources/aws-sdk-resources.gemspec#L180

But current status is when installing aws-sdk, aws-sdk-sso is never fetched. Is this some caching problem? The gem is published.

@dwradcliffe We currently don't have the bandwidth, especially several weeks out from re:Invent..
@simi AFAIK we use awscloud for service gems.

**Comment by JasonWainwright777** [comment]
This appears to still be an issue. Not sure if the mirrors have it cached or what, but we are still getting the errors.
Fetching aws-sdk-resources 3.57.0
Downloading aws-sdk-resources-3.57.0 revealed dependencies not in the API or the
lockfile (aws-sdk-sso (~> 1)).
Either installing with `--full-index` or running `bundle update
aws-sdk-resources` should fix the problem.
In Gemfile:
  aws-sdk was resolved to 3.0.1, which depends on
    aws-sdk-resources
STDERR:
---- End output of bundle install ----
Ran bundle install returned 34

**Comment by mullermp** [comment]
I have yanked aws-sdk-resources 3.57.0.

Please try reinstalling and trying gem clean if necessary.

**Comment by dabutvin** [comment]
this is now working for us! Thanks
